### PR TITLE
feat: Implement editing of device details

### LINKED ISSUE #2312: Implement editing of datas related to connected devices
**Describe the feature you'd like**
Implement the editing of data related to connected devices. Like change room name, Device name, location, etc.

**Additional context**
Users can change details of their connected devices in the web client.

**Screenshots**
Screenshot 2019-07-21 at 11 25 10 AM

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by atm1504**
Fixes #2312 

Changes: 
- [x] Store the macid
- [x] Add edit data fragment
- [x] Implement editing of rooms
- [x] Implement editing of speaker name
- [x] Update data on the server
- [x] Enhance the UI to make the flow smooth

Screenshots for the change: 
ezgif com-video-to-gif

### CODE REVIEW

File: app/src/main/java/org/fossasia/susi/ai/device/connecteddevices/ConnectedDeviceFragment.kt [hunk 1]
```diff
@@ -20,6 +20,7 @@ class ConnectedDeviceFragment : Fragment(), IConnectedDeviceView {
     private lateinit var connectedDevicePresenter: IConnectedDevicePresenter
     private var connectedDevicesAdapter: RecyclerView.Adapter<*>? = null
     private val connectedDeviceList: ArrayList<Device> = ArrayList()
+    private val macIdList: ArrayList<String> = ArrayList()
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/device/connecteddevices/ConnectedDeviceFragment.kt]:
```suggestion
    private val macIdList = mutableList<String>()
```

**Comment by atm1504** [comment]
@liveHarshit @iamareebjamal please review it.

File: app/src/main/java/org/fossasia/susi/ai/device/connecteddevices/ConnectedDeviceFragment.kt [hunk 2]
```diff
@@ -61,7 +67,15 @@ class ConnectedDeviceFragment : Fragment(), IConnectedDeviceView {
...
         connectedDeviceRecyclerView.adapter = connectedDevicesAdapter
     }
-}
\ No newline at end of file
+
+    override fun viewDevice(positiion: Int) {
+        val intent = Intent(activity, DeviceActivity::class.java)
+        intent.putExtra(CONNECT_TO, TAG_VIEW_DEVICE_FRAGMENT)
+        intent.putExtra("deviceDetails", connectedDeviceList[positiion])
+        intent.putExtra("macId", macIdList[positiion])
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/susi/ai/device/connecteddevices/ConnectedDeviceFragment.kt]: Declare key as constant

File: app/src/main/java/org/fossasia/susi/ai/device/viewdevice/adapters/ShowRoomsAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import android.view.View
+import android.view.ViewGroup
+import kotlinx.android.synthetic.main.room_recycler_layout.view.delete_room
+import kotlinx.android.synthetic.main.room_recycler_layout.view.image_tick
+import kotlinx.android.synthetic.main.room_recycler_layout.view.room_text
+import org.fossasia.susi.ai.R
+import org.fossasia.susi.ai.device.viewdevice.ViewDeviceFragment
+import org.fossasia.susi.ai.device.viewdevice.contract.IViewDevicePresenter
+
+class ShowRoomsAdapter(private val availableRoomsList: ArrayList<ViewDeviceFragment.AvailableRoomsFormat>, val context: Context, private val viewDevicePresenter: IViewDevicePresenter) : RecyclerView.Adapter<ShowRoomsAdapter.ViewHolder>() {
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/susi/ai/device/viewdevice/adapters/ShowRoomsAdapter.kt]: Too long. Separate in different lines

**Comment by atm1504** [comment]
@liveHarshit @iamareebjamal please review it. I have made the suggested changes.

**Comment by atm1504** [comment]
@liveHarshit @iamareebjamal please review it.

File: app/src/main/java/org/fossasia/susi/ai/device/viewdevice/ViewDevicePresenter.kt [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        var results = realm.where(RoomsAvailable::class.java).findAll()
+        var id = results.size
+        id++
+
+        var addedRoomModel = realm.createObject(RoomsAvailable::class.java)
+        addedRoomModel.id = id.toLong()
+        addedRoomModel.room = roomName
+        realm.commitTransaction()
+        viewDeviceView.showRooms()
+        // deviceConnectView?.roomNameSelected(null)
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/susi/ai/device/viewdevice/ViewDevicePresenter.kt:49]: Why?

**Review by liveHarshit — APPROVED** [review]

File: app/src/main/java/org/fossasia/susi/ai/device/viewdevice/ViewDevicePresenter.kt (hunk 4 above)
Comment (atm1504) [reply app/src/main/java/org/fossasia/susi/ai/device/viewdevice/ViewDevicePresenter.kt:49]: Actually, this statement was used for a feature to deselect previously selected options, when a new room is created. But now, I have removed that feature. So, it was commented. This feature can be re-added just by uncommenting it if required. Hence I kept that code line.

**Comment by atm1504** [comment]
@liveHarshit please review it

**Comment by liveHarshit** [comment]
> @liveHarshit please review it

I have already approved.

**Comment by atm1504** [comment]
@iamareebjamal please review it.

**Pull request merged**